pmda/util.py

```diff
@@ -30,3 +57,118 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.elapsed = end_time - self._start_time
         # always propagate exceptions forward
         return False
+
+
+def make_balanced_blocks(n_frames, n_blocks, start=None, step=None):
+    """Divide `n_frames` into `n_blocks` balanced blocks.
+
+    The blocks are generated in such a way that they contain equal numbers of
+    frames when possible, but there are also no empty blocks (which can happen
+    with a naive distribution of ``ceil(n_frames/n_blocks)`` per block and a
+    remainder block).
+
+    Arguments
+    ---------
+    n_frames : int
+        number of frames in the trajectory (≥0). This must be the number of
+        frames *after* the trajectory has been sliced,
+        i.e. ``len(u.trajectory[start:stop:step])`` and `start` and `step` must
+        be provided as parameters.
+    n_blocks : int
+        number of blocks (>0)
+    start : int
+        first index of the trajectory (default is ``None``, which is
+        interpreted as "first frame", i.e., 0); see comments on `n_frames`.
+    step : int
+        step by which the trajectory is sliced (see description of `n_frames`);
+        the default is ``None`` which corresponds to ``step=1``.
```

Comment:
why `None` instead of just `step=1` as the default. `None` makes sense if the default is a mutable type (e.g. an empty list). For immutable types like int, float, strings I would use them directly